src/tests/src/solver/modelParser/testModelTranslator.cpp

```diff
@@ -0,0 +1,237 @@
+    auto& parameter2 = model.Parameters().at("param2");
+    BOOST_CHECK_EQUAL(parameter1.Name(), "param1");
+    BOOST_CHECK(parameter1.isTimeDependent());
+    BOOST_CHECK(!parameter1.isScenarioDependent());
+    BOOST_CHECK_EQUAL(parameter1.Type(), ObjectModel::ValueType::FLOAT);
+    BOOST_CHECK_EQUAL(parameter2.Name(), "param2");
+    BOOST_CHECK(!parameter2.isTimeDependent());
+    BOOST_CHECK(!parameter2.isScenarioDependent());
+    BOOST_CHECK_EQUAL(parameter2.Type(), ObjectModel::ValueType::FLOAT);
+}
+
+// Test library with models and variables
+BOOST_AUTO_TEST_CASE(test_library_models_with_variables)
+{
+    ModelParser::Library library;
+    ModelParser::Model model1{
+      .id = "model1",
+      .description = "description",
+      .parameters = {},
+      .variables = {{"var1", "7", "pmax", ModelParser::ValueType::BOOL},
+                    {"var2", "99999999.9999999", "vcost", ModelParser::ValueType::INTEGER}},
+      .ports = {},
+      .port_field_definitions = {},
+      .constraints = {},
+      .objective = "objectives"};
+    library.models = {model1};
+    ObjectModel::Library lib = ModelConverter::convert(library);
+    auto& model = lib.models().at("model1");
+    BOOST_REQUIRE_EQUAL(model.Variables().size(), 2);
+    auto& variable1 = model.Variables().at("var1");
+    auto& variable2 = model.Variables().at("var2");
+    BOOST_CHECK_EQUAL(variable1.Name(), "var1");
+    BOOST_CHECK_EQUAL(variable1.LowerBound().Value(), "7");
+    BOOST_CHECK_EQUAL(variable1.UpperBound().Value(), "pmax");
+    BOOST_CHECK_EQUAL(variable1.Type(), ObjectModel::ValueType::BOOL);
+    BOOST_CHECK_EQUAL(variable2.Name(), "var2");
+    BOOST_CHECK_EQUAL(variable2.LowerBound().Value(), "99999999.9999999");
+    BOOST_CHECK_EQUAL(variable2.UpperBound().Value(), "vcost");
+    BOOST_CHECK_EQUAL(variable2.Type(), ObjectModel::ValueType::INTEGER);
+}
+
+// Test library with models and ports
+BOOST_AUTO_TEST_CASE(test_library_models_with_ports, *boost::unit_test::disabled())
+{
+    ModelParser::Library library;
+    ModelParser::Model model1{.id = "model1",
+                              .description = "description",
+                              .parameters = {},
+                              .variables = {},
+                              .ports = {{"port1", "flow"}, {"port2", "impedance"}},
+                              .port_field_definitions = {},
+                              .constraints = {},
+                              .objective = "objectives"};
+    library.models = {model1};
+    ObjectModel::Library lib = ModelConverter::convert(library);
+    auto& model = lib.models().at("model1");
+    // BOOST_REQUIRE_EQUAL(model.Ports().size(), 2);
+    // auto& port1 = model.Ports().at("port1");
+    // auto& port2 = model.Ports().at("port2");
+    // BOOST_CHECK_EQUAL(port1.Name(), "port1");
+    //  BOOST_CHECK_EQUALS port1.Type()
+    // BOOST_CHECK_EQUAL(port2.Name(), "port2");
+    // BOOST_CHECK_EQUALS port2.Type()
```

Comment:
This test does nothing

Reply:
Yes. *boost::unit_test::disabled()
It's here as a skeleton for when ports will be implemented properly
Can be deleted if you prefer so

Reply:
No, you may keep it here